renderlib/Colormap.cpp

```diff
@@ -0,0 +1,499 @@
+#include "Colormap.h"
+
+#include <QColor>
+
+#include <vector>
+#include <array>
+
+uint8_t*
+colormapFromColormap(uint8_t* colormap, size_t length)
+{
+  // basically just copy the whole thing.
+  uint8_t* lut = new uint8_t[length * 4]{ 0 };
+  for (size_t x = 0; x < length; ++x) {
+    lut[x * 4 + 0] = colormap[x * 4 + 0];
+    lut[x * 4 + 1] = colormap[x * 4 + 1];
+    lut[x * 4 + 2] = colormap[x * 4 + 2];
+    lut[x * 4 + 3] = colormap[x * 4 + 3];
+  }
+  return lut;
+}
+
+uint8_t*
+colormapFromControlPoints(std::vector<ColorControlPoint> pts, size_t length)
+{
+  // pts is piecewise linear from first to last control point.
+  // pts is in order of increasing x value (the first element of the pair)
+  // pts[0].first === 0
+  // pts[pts.size()-1].first === 1
```

Comment:
Remove?

Reply:
What happens if the first control point has `p[0]` set to a value other than `0`? Currently it looks like it would be initialized as a transparent black

Reply:
I added error checks